src/network/network_gaming_messages.cc

```diff
@@ -39,72 +45,45 @@ const std::string NetworkGamingMessages::get_message(const std::string& code,
                                                      const std::string& arg1,
                                                      const std::string& arg2,
                                                      const std::string& arg3) {
+
 	if (ngmessages.find(code) == ngmessages.end()) {
 		return (
 		   format("%s, %s, %s, %s", code, get_message(arg1), get_message(arg2), get_message(arg3)));
 	}
 
-	// push code and all arguments - if existing - in a vector for easier handling
-	std::vector<std::string> strings;
-	strings.push_back(get_message(code));
-	if (!arg1.empty()) {
-		strings.push_back(get_message(arg1));
-		if (!arg2.empty()) {
-			strings.push_back(get_message(arg2));
-			if (!arg3.empty()) {
-				strings.push_back(get_message(arg3));
-			}
-		}
-	}
+	assert(ngmessages[code].n_arg <= 3);
 
-	uint8_t last = strings.size() + 1;
-	// Try merging the strings
-	while (strings.size() > 1 && last > strings.size()) {
-		last = strings.size();
+	// for easier handling
+	std::vector<std::string> args;
+	args.push_back(arg1);
+	args.push_back(arg2);
+	args.push_back(arg3);
```

Comment:
Pull this all into one initializer list, and if possible to not copy the strings by using references (or pointers if that doesn't work)
```suggestion
	const std::string& args[] = {arg1, arg2, arg3};
```
```suggestion
	const std::string* args[] = {&arg1, &arg2, &arg3};
```

Reply:
References don't work, it must be pointers.